### PR TITLE
* Workaround for Issue 521, remove createTosaToStandard from Passes.c…

### PR DESCRIPTION
**Pull request by anupgangwar**
…pp and

  disable ElementwisePowModule_basic

Signed-off-by: Anup Gangwar <anup.gangwar@arm.com>

### CODE REVIEW

**Comment by anupgangwar** [comment]
Tagging @sjarus for merging.

**Comment by sjarus** [comment]
@cathyzhyi @ramiro050 any idea where the NllLossModule_basic RefBackend failure is coming from here ? The TOSA changes do not touch the ref backend.

**Comment by ramiro050** [comment]
> @cathyzhyi @ramiro050 any idea where the NllLossModule_basic RefBackend failure is coming from here ? The TOSA changes do not touch the ref backend.

The failure is because of a recent backwards-compatibility-breaking change made in PyTorch regarding the way `torch.ops.aten...` functions are handled. See this commit https://github.com/pytorch/pytorch/commit/043e84b3d2cb5affee6373ebc95cbafb82fe0017

I managed to fix the error on my machine by removing the `self=` code from these two tests

https://github.com/llvm/torch-mlir/blob/d69d29b7a65b7fa6a46c8cf49efeaabad4ecbb94/e2e_testing/torchscript/nll_loss.py#L28

https://github.com/llvm/torch-mlir/blob/d69d29b7a65b7fa6a46c8cf49efeaabad4ecbb94/e2e_testing/torchscript/nll_loss.py#L53

@anupgangwar, can you try adding this change to your PR to see if it fixes the issue?

**Comment by ramiro050** [comment]
There also seems to be an issue happening with the refbackend in the CI when printing the error messages. For reference, this is the error that the refbackend was not able to print in the CI:

```shell
nexpected outcome summary:

****** Failed tests - 2 tests
    FAIL - "NllLossModule_basic"
        Compilation error: Traceback (most recent call last):
          File "/home/ramiroleal/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/torchscript/framework.py", line 302, in run_tests
            golden_trace = generate_golden_trace(test)
          File "/home/ramiroleal/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/torchscript/framework.py", line 292, in generate_golden_trace
            test.program_invoker(tracer, TestUtils())
          File "/home/ramiroleal/torch-mlir/e2e_testing/torchscript/nll_loss.py", line 62, in NllLossModule_ignore_index
            module.forward(tu.rand(2, 3), torch.tensor([0, 1]))
          File "/home/ramiroleal/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/torchscript/framework.py", line 272, in __call__
            output = self.__wrapped__(*args, **kwargs)
          File "/home/ramiroleal/torch-mlir/e2e_testing/torchscript/nll_loss.py", line 53, in forward
            return torch.ops.aten.nll_loss_forward(self=x,
        TypeError: __call__() got multiple values for argument 'self'

    FAIL - "NllLossModule_ignore_index"
        Compilation error: Traceback (most recent call last):
          File "/home/ramiroleal/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/torchscript/framework.py", line 302, in run_tests
            golden_trace = generate_golden_trace(test)
          File "/home/ramiroleal/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/torchscript/framework.py", line 292, in generate_golden_trace
            test.program_invoker(tracer, TestUtils())
          File "/home/ramiroleal/torch-mlir/e2e_testing/torchscript/nll_loss.py", line 62, in NllLossModule_ignore_index
            module.forward(tu.rand(2, 3), torch.tensor([0, 1]))
          File "/home/ramiroleal/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/torchscript/framework.py", line 272, in __call__
            output = self.__wrapped__(*args, **kwargs)
          File "/home/ramiroleal/torch-mlir/e2e_testing/torchscript/nll_loss.py", line 53, in forward
            return torch.ops.aten.nll_loss_forward(self=x,
        TypeError: __call__() got multiple values for argument 'self'


Summary:
    Failed: 2
```

**Comment by anupgangwar** [comment]
@ramiro050 Thanks, will try it out. The PyTorch bit explains why we are not observing this on the local machine.

File: e2e_testing/torchscript/nll_loss.py [hunk 1]
```diff
@@ -50,8 +49,7 @@ def __init__(self):
   ])
   # None of the index is ignored here, since the ignored index is out of bounds.
   def forward(self, x, y):
-    return torch.ops.aten.nll_loss_forward(self=x,
-                                           target=y,
+    return torch.ops.aten.nll_loss_forward(target=y,
```
Comment (ramiro050) [inline e2e_testing/torchscript/nll_loss.py:52]: oh, I should've made this clearer. I meant replacing `self=x` with `x`

Comment (anupgangwar) [reply e2e_testing/torchscript/nll_loss.py:52]: ok

**Comment by anupgangwar** [comment]
@sjarus, @ramiro050 Checks passed. Can one of you merge this PR?

**Comment by silvasean** [comment]
I suspect this can be fixed by adding the right dependency here: https://github.com/anupgangwar/torch-mlir/blob/4757f0963d4e63a35c5727859f12949f8674aaeb/lib/Dialect/TorchConversion/Transforms/CMakeLists.txt#L19

**Comment by anupgangwar** [comment]
Will add it there if we bring in the pass again.

**Comment by sjarus** [comment]
The reason it is preferable to do TosaToStandard at the beginning of the LinAlg conversion is that it ensures the emitted form here is TOSA domain and not TOSA+standard . This ensures that the reference model - which parses tosa.const - works properly when doing the unit tests .

**Pull request merged**